code.js

```diff
@@ -1,3 +1,31 @@
 function mergesort(array) {
+   
+    Arrleng = array.length;
+
+    for (size = 1; size < Arrleng; size = size * 2) {
+        for (leftStart = 0; leftStart < Arrleng - 1; leftStart = leftStart + 2 * size){
+            mid = Math.min(leftStart + size, Arrleng);
+            rightEnd = Math.min(leftStart + 2 * size, Arrleng);
+
+            left = leftStart;
+            right = mid;
+
+            while (left < right && right < rightEnd) {
+                if (array[left] <= array[right]) {
+                    left++;
+                } 
+                else {
+                    temp = array[right];
+                    for (index = right; index > left; index--) {
```

Comment:
This loop is where you're doing more work than normal mergesort.

Reply:
so since there is an extra loop it would be n^2 log n ?

Reply:
That sounds like it's going in the right direction.

Reply:
There is three for loops and 1 while loop so is there another instance of n i need to add?

Reply:
How many iterations do the loops do?

Reply:
Are you asking how many times the loops goes through the current array or add to the array?

Reply:
My point is that not all of the loops have the same complexity.

Reply:
in the worse case the last for loop would have to shift every element so that would be O(n), unless we are talking about a different loop than the one we originally started with.

Reply:
Ok, but you still need to fix the final complexity.

Reply:
I just updated it, is that better?